j5/backends/hardware/j5/serial.py

```diff
@@ -92,7 +93,14 @@ def read_serial_line(self, empty: bool = False) -> str:
                 "Is it correctly powered?",
             )
 
-        ldata = bdata.decode('utf-8')
+        try:
+            ldata = bdata.decode('utf-8')
+        except UnicodeDecodeError as e:
+            if empty:
+                logging.getLogger(__file__).error(f"{e} in {bdata!r}")
+                ldata = ''
+            else:
+                raise
```

Comment:
Nit: Think we can short circuit here
```suggestion
                return ''
            raise
```